opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/StructureTasksInteractor.java

```diff
@@ -0,0 +1,170 @@
+import static org.smartregister.family.util.DBConstants.KEY.DOB;
+import static org.smartregister.family.util.DBConstants.KEY.FIRST_NAME;
+import static org.smartregister.family.util.DBConstants.KEY.LAST_NAME;
+import static org.smartregister.family.util.DBConstants.KEY.MIDDLE_NAME;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.BUSINESS_STATUS;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.CAMPAIGN_ID;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.CODE;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.FOR;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.ID;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.NAME;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.STATUS;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.STRUCTURES_TABLE;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.STRUCTURE_ID;
+import static org.smartregister.reveal.util.Constants.DatabaseKeys.TASK_TABLE;
+import static org.smartregister.reveal.util.FamilyConstants.TABLE_NAME.FAMILY_MEMBER;
+
+/**
+ * Created by samuelgithengi on 4/12/19.
+ */
+public class StructureTasksInteractor extends BaseInteractor implements StructureTasksContract.Interactor {
+
+
+    private static final String TAG = StructureTasksInteractor.class.getName();
+    private AppExecutors appExecutors;
+
+    private SQLiteDatabase database;
+    private StructureTasksContract.Presenter presenter;
+    private StructureRepository structureRepository;
+
+    public StructureTasksInteractor(StructureTasksContract.Presenter presenter) {
+        this(presenter, RevealApplication.getInstance().getAppExecutors(), RevealApplication.getInstance().getRepository().getReadableDatabase(), RevealApplication.getInstance().getStructureRepository());
+    }
+
+    @VisibleForTesting
+    protected StructureTasksInteractor(StructureTasksContract.Presenter presenter, AppExecutors appExecutors,
+                                       SQLiteDatabase database, StructureRepository structureRepository) {
+        super(presenter);
+        this.presenter = presenter;
+        this.appExecutors = appExecutors;
+        this.database = database;
+        this.structureRepository = structureRepository;
+    }
+
+    @Override
+    public void findTasks(String structureId, String campaignId) {
+        appExecutors.diskIO().execute(() -> {
+            List<StructureTaskDetails> taskDetailsList = new ArrayList<>();
+            Cursor cursor = null;
+            try {
+                cursor = database.rawQuery(getStructureSelect(String.format("%s=? AND %s=?", FOR, CAMPAIGN_ID)),
+                        new String[]{structureId, campaignId});
+                while (cursor.moveToNext()) {
+                    taskDetailsList.add(readTaskDetails(cursor));
+                }
+
+                cursor.close();
```

Comment:
Could the cursor ever be null?

Reply:
Is there a reason it is being closed here anyway?

Reply:
Cursor cannot be null, since for it to be null, an exception occurred and then executing sqliteDatabase.rawQuery and it would go into catch block.

The reason the cursor is being closed is because we are recreating it when querying for member tasks